http3/headers.go

```diff
@@ -124,11 +124,14 @@ func requestFromHeaders(headerFields []qpack.HeaderField) (*http.Request, error)
 		}
 	} else if len(hdr.Path) == 0 || len(hdr.Authority) == 0 || len(hdr.Method) == 0 {
 		return nil, errors.New(":path, :authority and :method must not be empty")
+	} else if len(hdr.Protocol) > 0 {
+		return nil, errors.New(":protocol must be empty")
```

Comment:
This branch needs to run `if !isExtendedConnected`. I think this is given by the current logic, but it's a bit subtle. I'd prefer to have this is in a separate if statement.

Reply:
Fixed